table/tables/virtual_tables.go

```diff
@@ -0,0 +1,169 @@
+// Copyright 2017 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tables
+
+import (
+	"github.com/juju/errors"
+	"github.com/pingcap/tidb/context"
+	"github.com/pingcap/tidb/kv"
+	"github.com/pingcap/tidb/meta/autoid"
+	"github.com/pingcap/tidb/model"
+	"github.com/pingcap/tidb/table"
+	"github.com/pingcap/tidb/util/types"
+)
+
+// VirtualDataSource is used to extract data from the struct in memory.
+type VirtualDataSource interface {
+	// GetRows, do the actual job
+	GetRows(ctx context.Context) (fullRows [][]types.Datum, err error)
+	// Meta return the meta of table
+	Meta() *model.TableInfo
+	// Cols return the cols of table
+	Cols() []*table.Column
+}
+
+// VirtualTable stands for the fake table all its data is in the memory.
+// dataSource: the function to get rows
+// @TODO this table is almost the same as the infoschema tables, but we need to use it in performance schema.
+// @TODO So we have to move it here, sometimes we need to refactor the infoschema tables to decrease the multiplicity of the codes
+type VirtualTable struct {
+	dataSource VirtualDataSource
+}
+
+// CreateVirtualTable , as its name
```

Comment:
remove this comma

Reply:
this is the rule by golint, it's useless, but we have to leave it here